### ISSUE TITLE
Buildpack PWADevServer should gracefully fall back to localhost if host/cert fails

### ISSUE THREAD

**Issue by brendanfalkowski** [magento-research/pwa-studio#679]
## This issue is for the following packages:

- [x] `venia-concept`
- [x] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`

## This issue is a:

- [x] Bug
- [ ] Feature suggestion
- [ ] Documentation issue
- [ ] Other (Please Specify)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             | 2.3-develop |
| Operating System + version  | macOS 10.14.2 |
| node.js version (`node -v`) | 8.9.3 |
| npm version (`npm -v`)      | 6.3.0 |

## Description

:warning: **Updated issue 2019/1/9 to reflect the underlying problem. Thanks, @brendanfalkowski, we're working on removing obstacles to getting the dev cycle started.**

I have a working Magento 2.3 instance using the `2.3-develop` branch, and the corresponding sample data installed. It's running via Valet Plus on macOS Mojave.

For the first-run setting up PWA Studio, I chose to use the Magento Cloud backend to see if that worked following these steps:

```bash
git clone https://github.com/magento-research/pwa-studio.git
cd pwa-studio
npm install
cp packages/venia-concept/.env.dist packages/venia-concept/.env
npm run build
npm run watch:venia
```

If fails on the last command with this error:

```bash
> @magento/pwa-studio@2.0.0 watch:venia /Users/brendan/work/repos/pwa-studio
> cd packages/venia-concept && npm run -s watch; cd - >/dev/null

Using environment variables from .env
Creating a local development domain requires temporary administrative privileges.
Please enter the password for brendan on Aragorn.local.
Password:
/bin/sh: /usr/local/Cellar/nss/3.39/bin/certutil: No such file or directory
Error: [pwa-buildpack:Utilities:configureHost.js] Could not setup development domain:
Error: Command failed: /usr/local/Cellar/nss/3.39/bin/certutil -A -d "/Users/brendan/Library/Application Support/Firefox/Profiles/ofr3i3tg.dev-edition-default" -t 'C,,' -i /var/folders/95/kr_8vnbn4y3_9hn30yj_2n4h0000gn/T/tmp-2180790TA8ba37qjW.tmp -n devcert
/bin/sh: /usr/local/Cellar/nss/3.39/bin/certutil: No such file or directory

    at checkExecSyncError (child_process.js:601:13)
    at Object.execSync (child_process.js:641:13)
    at Object.run (/Users/brendan/work/repos/pwa-studio/node_modules/devcert/dist/utils.js:24:28)
    at glob_1.sync.forEach (/Users/brendan/work/repos/pwa-studio/node_modules/devcert/dist/platforms/shared.js:28:25)
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/Users/brendan/work/repos/pwa-studio/node_modules/devcert/dist/platforms/shared.js:24:33)
    at Generator.next (<anonymous>)
    at /Users/brendan/work/repos/pwa-studio/node_modules/tslib/tslib.js:107:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/brendan/work/repos/pwa-studio/node_modules/tslib/tslib.js:103:16)
    at Object.addCertificateToNSSCertDB (/Users/brendan/work/repos/pwa-studio/node_modules/devcert/dist/platforms/shared.js:22:20)
    at MacOSPlatform.<anonymous> (/Users/brendan/work/repos/pwa-studio/node_modules/devcert/dist/platforms/darwin.js:51:32)
    at Generator.next (<anonymous>)
    at fulfilled (/Users/brendan/work/repos/pwa-studio/node_modules/tslib/tslib.js:104:62)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at configureHost (/Users/brendan/work/repos/pwa-studio/packages/pwa-buildpack/dist/Utilities/configureHost.js:123:15)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by mhhansen** [comment]
looks like it fails to add the certificate into your nss db (can't double check, don't have installed it).
but seems that the `brew list` throws a hit (or a false positive, due to a bug in that validation) so it goes for that option...

try this [devcert PR fix](https://github.com/davewasmer/devcert/pull/30) out:
```
# node_modules/devcert/dist/platforms/darwin.js :: 86
# before:
return utils_1.run('brew list').toString().indexOf('nss') > -1;

# after:
return utils_1.run('brew list').toString().split('\n').indexOf('nss') > -1;
```

cause the path that's failing is being built based on the assumption that you have `nss` (which you may not)
```
# node_modules/devcert/dist/platforms/darwin.js -- line 49
let certutilPath = path_1.default.join(utils_1.run('brew --prefix nss').toString().trim(), 'bin', 'certutil');
yield shared_1.closeFirefox();
yield shared_1.addCertificateToNSSCertDB(this.FIREFOX_NSS_DIR, certificatePath, certutilPath);

# in your case, `certutilPath` ends being:
/usr/local/Cellar/nss/3.39/bin/certutil
```

not sure why I never got this error (I think I've your same local setup)

**Comment by brendanfalkowski** [comment]
@mhhansen — Thanks for that idea — got it working again! I tried running `brew install nss` first, and that allowed `npm run watch:venia` to build. I'm wondering if me recently updating to macOS Mojave broke something symlinked by `brew` on my machine.

I tried testing your original approach by running `brew uninstall nss` and fixing the "devcert" bug's PR manually, but I couldn't get `npm run watch:venia` command to fail again. Not what I expected but `brew` always looks opaque to me.

I'll bump that PR with this discussion.
